sql/catalyst/src/test/scala/org/apache/spark/sql/catalyst/expressions/MathExpressionsSuite.scala

```diff
@@ -199,6 +199,12 @@ class MathExpressionsSuite extends SparkFunSuite with ExpressionEvalHelper {
     checkConsistencyBetweenInterpretedAndCodegen(Sinh, DoubleType)
   }
 
+  test("asinh") {
+    testUnary(Asinh, (x: Double) => math.log(x + math.sqrt(x * x + 1.0)))
+    checkConsistencyBetweenInterpretedAndCodegen(Asinh, DoubleType)
+    checkEvaluation(Asinh(Double.NegativeInfinity), Double.NegativeInfinity)
```

Comment:
can you also add test cases passing `null` as input?

Reply:
+1 for @mgaido91 's advice.

Reply:
The input type of these math functions is `Double`, which is a value type and I don't think scala compiler allows assigning `null` to a `Double`.

Reply:
in SQL you can pass null. You can have a `null` Double literal for instance. Moreover, yes, in java a `Double` can be `null`. In your code you haven't managed it, because null handling is performed in `defineCodeGen`. But having tests for it enforces that there won't be regressions.

Reply:
@mgaido91 thanks for sharing. Now added null test cases for asinh/acos/atanh - referring test code for `cot`.